msgraphforentra_connector.py

```diff
@@ -0,0 +1,1631 @@
+        if app_connector:
+            app_connector.debug_print("In _save_app_state: Invalid asset_id")
+        return {}
+
+    if app_connector:
+        app_connector.debug_print("Saving state: ", state)
+
+    try:
+        with open(real_state_file_path, "w+") as state_file_obj:
+            state_file_obj.write(json.dumps(state))
+    except Exception as e:
+        print("Unable to save state file: {0}".format(str(e)))
+
+    return phantom.APP_SUCCESS
+
+
+def _handle_login_response(request):
+    """This function is used to get the login response of authorization request from Microsoft login page.
+
+    :param request: Data given to REST endpoint
+    :return: HttpResponse. The response displayed on authorization URL page
+    """
+
+    asset_id = request.GET.get("state")
+    if not asset_id:
+        return HttpResponse("ERROR: Asset ID not found in URL\n{}".format(json.dumps(request.GET)), content_type="text/plain", status=400)
+
+    # Check for error in URL
+    error = request.GET.get("error")
+    error_description = request.GET.get("error_description")
+
+    # If there is an error in response
+    if error:
+        message = "Error: {0}".format(error)
+        if error_description:
+            message = "{0} Details: {1}".format(message, error_description)
+        return HttpResponse("Server returned {0}".format(message), content_type="text/plain", status=400)
+
+    code = request.GET.get(MSGENTRA_CODE_STRING)
+
+    # If code is not available
+    if not code:
+        return HttpResponse("Error while authenticating\n{0}".format(json.dumps(request.GET)), content_type="text/plain", status=400)
+
+    state = _load_app_state(asset_id)
+
+    # If value of admin_consent is not available, value of code is available
+    try:
+        state[MSGENTRA_CODE_STRING] = code
+        state[MSGENTRA_STATE_IS_ENCRYPTED] = True
+    except Exception as e:
+        return HttpResponse("{}: {}".format(MSGENTRA_DECRYPTION_ERROR, str(e)), content_type="text/plain", status=400)
+
+    _save_app_state(state, asset_id, None)
+
+    return HttpResponse("Code received. Please close this window, the action will continue to get new token.", content_type="text/plain")
+
+
+def _get_dir_name_from_app_name(app_name):
+    """Get name of the directory for the app.
+
+    :param app_name: Name of the application for which directory name is required
+    :return: app_name: Name of the directory for the application
+    """
+
+    app_name = "".join([x for x in app_name if x.isalnum()])
+    app_name = app_name.lower()
+    if not app_name:
+        app_name = "app_for_phantom"
+    return app_name
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class MsGraphForEntra_Connector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(MsGraphForEntra_Connector, self).__init__()
```

Comment:
Nit: this is py2-ism
```suggestion
        super().__init__()
```